src/ucs/sys/topo/base/topo.c

```diff
@@ -790,3 +790,51 @@ const char *ucs_topo_resolve_sysfs_path(const char *dev_path, char *path_buffer)
               sysfs_path);
     return sysfs_path;
 }
+
+static ucs_status_t
+uct_topo_iface_parse_virtual_dev(const struct dirent *entry, void *ctx)
+{
+    static const char *low_level_dev_prefix = "lower_";
+    ucs_string_buffer_t *dev_path           = ctx;
+
+    if (!strncmp(entry->d_name, low_level_dev_prefix,
+                 strlen(low_level_dev_prefix))) {
+        ucs_string_buffer_appendf(dev_path, "/%s", entry->d_name);
+        return UCS_ERR_CANCELED;
+    }
+
+    return UCS_OK;
+}
+
+const char *ucs_topo_iface_lowest_device_path(const char *dev_name,
```

Comment:
`static const char *ucs_topo_iface_lowest_device_path`

Reply:
Looks it cannot be a static function since it's used by both tcp_iface.c(uct_tcp_iface_get_sysfs_path) and sock.c(ucs_netif_bond_ad_num_ports)